src/main/java/org/elasticsearch/action/admin/indices/segments/IndicesSegmentResponse.java

```diff
@@ -139,6 +142,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         if (segment.getMergeId() != null) {
                             builder.field(Fields.MERGE_ID, segment.getMergeId());
                         }
+                        if (segment.ramTree != null) {
+                            builder.startArray(Fields.RAM_TREE);
+                            // the root Accountable is just a holder for the top level elements of the segment
+                            Iterator<Accountable> itr = segment.ramTree.getChildResources().iterator();
```

Comment:
It was trying to guard against an empty iteration (ie bug in what is put in ramtree to be an empty iterable).  But I'm fine with using a for each here.